torch/quantization/observer.py

```diff
@@ -9,44 +9,50 @@
 import torch.nn as nn
 from torch._jit_internal import List, Optional
 
+def _with_args(cls_or_self, **kwargs):
+    """Wrapper that allows creation of class factories.
 
-class _PartialWrapper(object):
-    def __init__(self, p):
-        self.p = p
-
-    def __call__(self, *args, **keywords):
-        return self.p(*args, **keywords)
-
-    def __repr__(self):
-        return self.p.__repr__()
+    This can be useful when there is a need to create classes with the same
+    constructor arguments, but different instances.
 
+    .. Example::
 
-def _with_args(cls_or_self, **kwargs):
+        >>> Foo.with_args = classmethod(_with_args)
+        >>> foo_builder = Foo.with_args(a=3, b=4).with_args(answer=42)
+        >>> foo_instance1 = foo_builder()
+        >>> foo_instance2 = foo_builder()
+        >>> id(foo_instance1) == id(foo_instance2)
+        False
     """
-    Wrapper around functools.partial that allows chaining.
+    class _PartialWrapper(object):
+        def __init__(self, p):
+            self.p = p
 
-    Often you want to assign it to a class as a class method:
+        def __call__(self, *args, **keywords):
+            return self.p(*args, **keywords)
 
-        Foo.with_args = classmethod(_with_args)
-        Foo.with_args(x=1).with_args(y=2)
-    """
+        def __repr__(self):
+            return self.p.__repr__()
+
+        with_args = _with_args
     r = _PartialWrapper(partial(cls_or_self, **kwargs))
     return r
 
-_PartialWrapper.with_args = _with_args
-
 
 ABC = ABCMeta(str("ABC"), (object,), {})  # compatible with Python 2 *and* 3:
 
 
 class Observer(ABC, nn.Module):
```

Comment:
Do we want to fix the name for Observer as well?

Reply:
I remember we want to rename this to `ObserverBase`

Reply:
That will be part of a different PR.